frontend/src/App.jsx

```diff
@@ -14,6 +14,9 @@ import RegisterPage from './pages/RegisterPage';
 import { IdeaAddPage, IdeaEditPage, IdeaPage, IdeasPage } from './pages/Ideas';
 import './styles.css';
 
+import ErrorBoundary from './components/ErrorBoundary.jsx';
+import NotFound from './pages/NotFound.jsx';
+
```

Comment:
Let's try to keep these two PR separate. There's no `ErrorBoundary` or `NotFound` in this branch.

Routes below need update as well.